src/library/dlgtrackinfo.cpp

```diff
@@ -239,6 +283,22 @@ void DlgTrackInfo::loadTrack(TrackPointer pTrack) {
             &DlgTrackInfo::slotTrackChanged);
 }
 
+void DlgTrackInfo::loadTrack(TrackPointer pTrack) {
+    VERIFY_OR_DEBUG_ASSERT(!m_pTrackModel) {
+        return;
+    }
+    changeTrack(pTrack);
```

Comment:
Likewise with DlgTagFetcher, this function is redundant with `changeTrack`. Pick one name and delete the other function.

Reply:
I had to divide it to set the check on `m_pTrackModel`.
A `trackModel` might be supplied in the constructor, but the overloaded `loadTrack(TrackPointer)` function might be called instead of `loadTrack(QModelIndex)` which will break next/previous buttons.
The `loadTrack` functions do the checks and `changeTrack` actually completes the loading of the track if everything is correct.
Would you like to suggest a change in name for the `changeTrack` function?

Reply:
Move the code from `changeTrack` into `loadTrack(TrackPointer)`, then the `loadTrack(QModelIndex)` overload can call `loadTrack(TrackPointer)`.

Reply:
This can lead to unexpected behaviour.
We'll need to remove `VERIFY_OR_DEBUG_ASSERT(!m_pTrackModel)` from `loadTrack(TrackPointer)` which makes sure this function is only used when a trackModel is not set in constructor. If we keep the assertion, `loadTrack(QModelIndex)` can't use this function. But if we remove the assertion, a client might erroneously set the trackModel in constructor and also use `loadTrack(TrackPointer)` instead of `loadTrack(QModelIndex)` which is required for the functioning of next/prev buttons.

I we are to remove the `changeTrack(TrackPointer)` function, the assertion in `loadTrack(TrackPointer)` will be changed to
```cpp
VERIFY_OR_DEBUG_ASSERT(!m_pTrackModel || m_pTrackModel->getTrack(m_currentTrackIndex) == pTrack) {
    return;
}

Reply:
> We'll need to remove VERIFY_OR_DEBUG_ASSERT(!m_pTrackModel) from loadTrack(TrackPointer) which makes sure this function is only used when a trackModel is not set in constructor.

Oh, good point. Then let's rename `changeTrack` to `loadTrackInternal` or `loadTrackPrivate`.